### PR TITLE
Fail CI builds with a better error when APIs are unexpectedly changed

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR adds a new CI-specific npm script, `npm run build:docs:ci`, that will fail the build with a (somewhat) more helpful error like this:

> `@fluidframework/driver-utils: Warning: You have changed the public API signature for this project. Please copy the file "_api-extractor-temp/driver-utils.api.md" to "E:\code\FluidFramework\api-report\driver-utils.api.md", or perform a local build (which does this automatically). See the Git repo documentation for more info.`

Running `npm run build:docs` locally will update the API report and output a warning like this:

> `@fluidframework/driver-utils: Warning: You have changed the public API signature for this project. Updating E:\code\FluidFramework\api-report\driver-utils.api.md`

### CODE REVIEW

File: api-report/driver-utils.api.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ export function combineAppAndProtocolSummary(appSummary: ISummaryTree, protocolS
 export function configurableUrlResolver(resolversList: IUrlResolver[], request: IRequest): Promise<IResolvedUrl | undefined>;
 
 // @public (undocumented)
-export function createGenericNetworkError(errorMessage: string, canRetry: boolean, retryAfterSeconds?: number, statusCode?: number): GenericNetworkError | ThrottlingError;
+export function createGenericNetworkError(errorMessage: string, canRetry: boolean, retryAfterSeconds?: number, statusCode?: number): ThrottlingError | GenericNetworkError;
```
Comment (tylerbutler) [inline api-report/driver-utils.api.md:115]: I don't know what caused this.

File: packages/drivers/odsp-driver/package.json [hunk 2]
```diff
@@ -13,8 +13,9 @@
   "scripts": {
     "build": "npm run build:genver && concurrently npm:build:compile npm:lint",
     "build:commonjs": "npm run tsc && npm run build:test",
-    "build:compile": "concurrently npm:build:commonjs npm:build:esnext",
+    "build:compile": "concurrently npm:build:commonjs npm:build:esnext && npm run build:docs",
```
Comment (anthony-murphy) [inline packages/drivers/odsp-driver/package.json:16]: we shouldn't build docs in the compile step. is can be in the build script

**Comment by anthony-murphy** [comment]
how does this fail the build? its not obvjous to me

File: packages/drivers/odsp-driver/package.json (hunk 2 above)
Comment (tylerbutler) [reply packages/drivers/odsp-driver/package.json:16]: Good catch; this was a mistake.

**Comment by tylerbutler** [comment]
> how does this fail the build? its not obvious to me

See this failure for an example: https://dev.azure.com/fluidframework/public/_build/results?buildId=23606&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=c32d47e1-65cf-544f-3c15-66e3d2029546

**Comment by anthony-murphy** [comment]
The example is useful, but i still don't see how it does it, like what is the mechanism, build:docs:ci looks to be the same a build doc, but somehow that one fails the build

File: server/routerlicious/packages/protocol-definitions/package.json [hunk 3]
```diff
@@ -11,10 +11,10 @@
   "module": "lib/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "build": "concurrently npm:build:compile npm:lint",
+    "build": "concurrently npm:build:compile npm:lint && npm run build:docs",
     "build:compile": "concurrently npm:tsc npm:build:esnext",
     "build:docs": "api-extractor run --local && copyfiles -u 1 ./_api-extractor-temp/doc-models/* ../../_api-extractor-temp/",
-    "build:docs:ci": "api-extractor run",
+    "build:docs:ci": "api-extractor run && copyfiles -u 1 ./_api-extractor-temp/doc-models/* ../../../_api-extractor-temp/",
```
Comment (anthony-murphy) [inline server/routerlicious/packages/protocol-definitions/package.json:17]: oh. is it just the lack of the --local right here?

**Comment by tylerbutler** [comment]
> The example is useful, but i still don't see how it does it, like what is the mechanism, build:docs:ci looks to be the same a build doc, but somehow that one fails the build

I misunderstood your question. Sorry! The only difference between the docs CI and the local docs builds is that the `--local` flag is passed when building locally. That flag causes api-extractor to automatically update the API report rather than failing the build with an error. From the api-extractor docs on the flag:

> Indicates that API Extractor is running as part of a local build, e.g. on a developer's machine. This disables certain validation that would normally be performed for a ship/production build. For example, the *.api.md report file is automatically copied in a local build.

File: packages/dds/map/src/interfaces.ts [hunk 4]
```diff
@@ -322,7 +322,6 @@ export interface ISerializedValue {
  * value is whatever params the ValueType needs to complete that operation.  Similar to ISerializableValue, it is
  * serializable via JSON.stringify/parse but differs in that it has no equivalency with an in-memory value - rather
  * it just describes an operation to be applied to an already-in-memory value.
- * @alpha
  */
 export interface IValueTypeOperationValue {
```
Comment (tylerbutler) [inline packages/dds/map/src/interfaces.ts:326]: @ChumpChief With the deprecation and removal of ValueTypes, can this interface be completely removed?

Comment (ChumpChief) [reply packages/dds/map/src/interfaces.ts:326]: Unfortunately no - Sequence still uses MapKernel to use valuetypes and we don't have a plan in place to change that.

Comment (tylerbutler) [reply packages/dds/map/src/interfaces.ts]: Yuck. OK. We'll make sense of this as part of API review. :)

**Pull request merged**